src/alfasim_sdk/result_reader/aggregator.py

```diff
@@ -85,6 +91,124 @@ class ResultsNeedFullReloadError(RuntimeError):
+    :ivar position_unit:
+        The position unit (m, Km, cm).
+    :ivar unit:
+        The unit of a dimensionless property (-, %)
+    :ivar qoi_index:
+        The index of a quantity of interest in the results file.
+    :ivar qoi_data_index:
+        The data index of a quantity of interest.
+    """
+
+    property_id: str = attr.ib(validator=attr.validators.instance_of(str))
+    trend_id: str = attr.ib(validator=attr.validators.instance_of(str))
+    category: str = attr.ib(validator=attr.validators.instance_of(str))
+    parametric_var_id: str = attr.ib(validator=attr.validators.instance_of(str))
+    parametric_var_name: str = attr.ib(validator=attr.validators.instance_of(str))
+    network_element_name: Optional[str] = attr.ib(
+        validator=attr.validators.optional(attr.validators.instance_of(str))
+    )
+    position: Optional[float] = attr.ib(
+        validator=attr.validators.optional(attr.validators.instance_of(float))
+    )
+    position_unit: Optional[str] = attr.ib(
+        validator=attr.validators.optional(attr.validators.instance_of(str))
+    )
+    unit: str = attr.ib(validator=attr.validators.instance_of(str))
+    qoi_index: Optional[int] = attr.ib(
+        validator=attr.validators.optional(attr.validators.instance_of(int))
+    )
+    qoi_data_index: Optional[int] = attr.ib(
+        validator=attr.validators.optional(attr.validators.instance_of(int))
+    )
+
+
+@attr.s(slots=True, hash=False)
+class GlobalSensitivityAnalysisMetadata:
+    """
+    A class that hold the global sensitivity analysis metadata.
+    """
+
+    global_sensitivity_analysis_meta_data: Dict[
+        str, GlobalSensitivityAnalysisMetaInformation
+    ] = attr.ib(validator=attr.validators.instance_of(Dict))
+    result_directory: Path = attr.ib(validator=attr.validators.instance_of(Path))
+
+    @classmethod
+    def empty(cls, result_directory: Path) -> "GlobalSensitivityAnalysisMetadata":
+        return GlobalSensitivityAnalysisMetadata(
+            global_sensitivity_analysis_meta_data={}, result_directory=result_directory
+        )
+
+    @classmethod
+    def from_dict(
+        cls, data: dict[str, Any]
+    ) -> "GlobalSensitivityAnalysisMetaInformation":
+        return GlobalSensitivityAnalysisMetaInformation(
+            property_id=data["property_id"],
+            trend_id=data["trend_id"],
+            category=data["category"],
+            parametric_var_id=data["parametric_var_id"],
+            parametric_var_name=data["parametric_var_name"],
+            network_element_name=data["network_element_name"],
+            position=data["position"],
+            position_unit=data["position_unit"],
+            unit=data["unit"],
+            qoi_index=data["qoi_index"],
+            qoi_data_index=data["qoi_data_index"],
+        )
+
+    @classmethod
+    def get_metadata_from_dir(
+        cls, result_directory: Path
+    ) -> Optional["GlobalSensitivityAnalysisMetadata"]:
```

Comment:
```suggestion
    ) -> "GlobalSensitivityAnalysisMetadata":
```

Reply:
And as @prusse-martin has been pointed, the ```Self``` is not supoorted. I tryied to do this and this error occur:
```shell
ImportError while loading conftest '/home/users/gustavopb/Projects/alfasim-sdk/tests/conftest.py'.
../conftest.py:12: in <module>
    from alfasim_sdk.result_reader.aggregator_constants import META_GROUP_NAME, GLOBAL_SENSITIVITY_ANALYSIS_GROUP_NAME, \
../../src/alfasim_sdk/result_reader/__init__.py:1: in <module>
    from .aggregator import ALFASimResultMetadata
../../src/alfasim_sdk/result_reader/aggregator.py:132: in <module>
    class GlobalSensitivityAnalysisMetadata:
../../src/alfasim_sdk/result_reader/aggregator.py:161: in GlobalSensitivityAnalysisMetadata
    def get_metadata_from_dir(cls, result_directory: Path) -> self:
E   NameError: name 'self' is not defined
```